pkg/controllers/cassandra/nodepool/resource.go

```diff
@@ -94,6 +94,14 @@ func StatefulSetForCluster(
 									jolokiaPort,
 									jolokiaContext,
 								),
+								fmt.Sprintf(
+									"--datacenter-node-label=%s",
+									np.FailureZones.Datacenter.ValueFrom.NodeLabel,
```

Comment:
We should actually check this exists, and pick between `Value`, or one of the potential `ValueFrom`s